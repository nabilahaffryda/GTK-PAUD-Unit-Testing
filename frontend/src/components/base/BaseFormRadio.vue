<template>
  <div>
    <v-subheader :class="[`px-0 body-2 ${labelColor}--text`]" style="height: 24px">
      <span :style="{ fontSize: fontSize, fontWeight: weight }">{{ label }}</span> {{ required ? '*' : '' }}
    </v-subheader>
    <v-radio-group
      v-model="form"
      :row="row !== undefined ? row : true"
      :class="margin || 'mt-0'"
      :error-messages="errorMessages"
    >
      <v-radio
        v-for="(item, i) in items"
        :key="i"
        :label="item.text"
        :value="item.value"
        :disabled="disabled"
      ></v-radio>
    </v-radio-group>
  </div>
</template>

<script>
export default {
  name: 'FormRadio',
  props: [
    'label',
    'labelColor',
    'required',
    'items',
    'value',
    'disabled',
    'row',
    'margin',
    'fontSize',
    'weight',
    'errorMessages',
  ],
  data() {
    return {
      form: this.value,
    };
  },
  methods: {
    mutate(value) {
      this.$emit('input', value);
    },

    setForm(value) {
      this.$set(this, 'form', value);
    },
  },
  watch: {
    form: 'mutate',
    value: 'setForm',
  },
};
</script>

<style scoped></style>
