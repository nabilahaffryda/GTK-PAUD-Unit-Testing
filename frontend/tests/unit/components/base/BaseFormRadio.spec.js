import { mount, createLocalVue } from "@vue/test-utils";
import BaseFormRadio from "@/components/base/BaseFormRadio.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseFormRadio.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('should show form radio', () => {
        const VRadio = {
            props: ['label',
                'labelColor',
                'required',
                'items',
                'value',
                'disabled',
                'row',
                'margin',
                'fontSize',
                'weight',
                'errorMessages',],
            template: '<div><slot :item="item" /></div>'
        }

        const wrapper = mount(BaseFormRadio, {
            localVue,
            vuetify,
            stubs: {
                VRadio
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
