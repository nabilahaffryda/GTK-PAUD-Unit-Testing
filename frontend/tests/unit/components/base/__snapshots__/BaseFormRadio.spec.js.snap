// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormRadio.vue should show form radio 1`] = `
<div>
  <div class="v-subheader theme--light px-0 body-2 undefined--text" style="height: 24px;"><span></span>
  </div>
  <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--row mt-0">
    <div class="v-input__control">
      <div class="v-input__slot" style="height: auto;">
        <div role="radiogroup" aria-labelledby="input-3" class="v-input--radio-group__input"></div>
      </div>
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
